lightspeed/modules/proc_generate-training-data-set.adoc 

```diff
@@ -0,0 +1,102 @@
+
+== Methods of generating a training data set
+You can generate a training data set by using one of the following methods:
+
+* *With ansible-lint preprocessing* 
++
+By default, the {AnsibleContentParser} generates training data sets by using ansible-lint preprocessing. The {AnsibleContentParser} uses ansible-lint rules to scan your Ansible files and ensure that the content adheres to Ansible best practices. If rule violations are found, the {AnsibleContentParser} excludes these files from the generated output. In such scenarios, you must resolve the rule violations, and run the {AnsibleContentParser} once again so that the generated output includes all your Ansible files. 
+
+* *Without ansible-lint preprocessing*
++
+You can specify to generate a training data set without ansible-lint preprocessing. In this method, the {AnsibleContentParser} does not scan your Ansible files for ansible-lint rule violations; therefore, the training data set includes all files. Although the training data set includes all files, it might not adhere to Ansible best practices and could affect the quality of your code recommendation experience. 
+
+.Prerequisites
+
+* You must have installed {AnsibleContentParser} on your computer.
+* You must have verified that the version of ansible-lint that is installed with the content parser tool is the same as that of the previously-installed ansible-lint.
+
+.Procedure
+
+. Run the {AnsibleContentParser} to generate a training data set: 
+** With ansible-lint preprocessing: `$ ansible-content-parser _source_ _output_`
+** Without ansible-lint preprocessing: `$ ansible-content-parser _source_ _output_ _-S_`
++
+The following table lists the required parameters.
++
+[cols="30%,70%",options="header"]
+|===
+|Parameter |Description
+|`_source_` 
+|Specifies the source of the training data set. 
+
+|`_output_`
+|Specifies the source of the training data set.
+
+|`_-S_` or `_--skip-ansible-lint_`
+|Specifies to skip ansible-lint preprocessing while generating the training data set.
+
+|===
+
++
+For example:
+If the source is a Github URL https://github.com/ansible/ansible-tower-samples.git, and the output directory is `/tmp/out`, the command prompt is as follows: +
+`$ ansible-content-parser https://github.com/ansible/ansible-tower-samples.git /tmp/out`
+
+. Optional: To generate a training data set with additional information, specify the following parameters while running the {AnsibleContentParser}.
++
+[cols="30%,70%",options="header"]
+|===
+|Parameter |Description
+
+|`--source-license`
+|Specifies to include the licensing information of the source directory in the training data set.
+
+|`--source-description`
+|Specifies to include the descriptions of the source directory in the training data set.
+
+|`--repo-name`
+|Specifies to include the repository name in the training data set. If you do not specify the  repository name, the {AnsibleContentParser} automatically generates it from the source name.
```

Comment:
Looks like a double space here: "If you do not specify the  repository name"

Reply:
Updated.